service/exporter/k8s/k8s.go

```diff
@@ -0,0 +1,386 @@
+func getDaemonSets(c *kubernetes.Clientset, ns string) ([]DaemonSet, error) {
+	dss, err := c.DaemonSets(ns).List(metav1.ListOptions{})
+
+	if err != nil {
+		return []DaemonSet{}, microerror.Mask(err)
+	}
+
+	ds := []DaemonSet{}
+	for _, s := range dss.Items {
+
+		ds = append(ds, DaemonSet{
+			metadata: metadata{
+				Labels: s.Labels,
+			},
+			Status:      s.Status,
+			Selector:    *s.Spec.Selector,
+			PodTemplate: fromPodTemplateSpec(s.Spec.Template),
+		})
+	}
+	return ds, nil
+}
+
+func getStatefulSets(c *kubernetes.Clientset, ns string) ([]StatefulSet, error) {
+
+	ss := []StatefulSet{}
+	statefulSets, err := c.AppsV1beta1().StatefulSets(ns).List(metav1.ListOptions{})
+
+	if err != nil {
+		return ss, microerror.Mask(err)
+	}
+
+	for _, s := range statefulSets.Items {
+
+		ss = append(ss, StatefulSet{
+			metadata: metadata{
+				Labels: s.Labels,
+			},
+			ServiceName: s.Spec.ServiceName,
+			Replicas:    *s.Spec.Replicas,
+			PodTemplate: fromPodTemplateSpec(s.Spec.Template),
+			Selector:    *s.Spec.Selector,
+			Status:      s.Status,
+		})
+	}
+
+	return ss, nil
+}
+
+func getCronJobs(c *kubernetes.Clientset, ns string) ([]CronJob, error) {
+
+	cj := []CronJob{}
+	cronjobs, err := c.BatchV2alpha1().CronJobs(ns).List(metav1.ListOptions{})
+
+	if err != nil {
+		return cj, microerror.Mask(err)
+	}
+
+	for _, s := range cronjobs.Items {
+		cj = append(cj, CronJob{
+			metadata: metadata{
+				Labels: s.Labels,
+			},
+			Name:        s.GetName(),
+			Schedule:    s.Spec.Schedule,
+			Status:      s.Status,
+			Suspend:     *s.Spec.Suspend,
+			JobTemplate: fromJobTemplateSpec(s.Spec.JobTemplate),
+		})
+	}
+	return cj, nil
+}
+
+func getIngresses(c *kubernetes.Clientset, ns string) ([]Ingress, error) {
+	is := []Ingress{}
+	ingresses, err := c.ExtensionsV1beta1().Ingresses(ns).List(metav1.ListOptions{})
+
+	if err != nil {
+		return is, microerror.Mask(err)
+	}
+	for _, i := range ingresses.Items {
+		is = append(is, Ingress{
+			metadata: metadata{
+				Labels: i.Labels,
+			},
+			Status:   i.Status,
+			Backends: MustBackend(i.Spec.Backend),
+			Rules:    i.Spec.Rules,
+			TLSHosts: i.Spec.TLS,
+		})
+	}
+
+	return is, nil
+}
+
+type NetworkPolicy struct {
+	metadata
+	Name         string
+	IngressRules []v1b1.NetworkPolicyIngressRule
+	Selector     metav1.LabelSelector
+}
+
+func GetNetworkPolicies(c *kubernetes.Clientset, ns string) ([]NetworkPolicy, error) {
+	nps := []NetworkPolicy{}
+	npl := &v1b1.NetworkPolicyList{}
+	err := c.ExtensionsV1beta1().RESTClient().Get().
+		Namespace(ns).
+		Resource("networkpolicies").
+		Do().
+		Into(npl)
+	if err != nil {
+		return []NetworkPolicy{}, microerror.Mask(err)
+	}
+
+	for _, n := range npl.Items {
+		nps = append(nps, NetworkPolicy{
+			metadata: metadata{
+				Labels: n.Labels,
+			},
+			Name:         n.Name,
+			IngressRules: n.Spec.Ingress,
+			Selector:     n.Spec.PodSelector,
+		})
+	}
+
+	return nps, nil
+}
+
+func MustBackend(b *v1b1.IngressBackend) v1b1.IngressBackend {
```

Comment:
Nitpick. AFAIK golang convention on naming functions with `must` wording usually imply panic on error. Here is no error.